### PR TITLE
Ensure user and personal org creation after successful sign up/in

### LINKED ISSUE #491: Unable to successfully launch GraphQL Hive with self-hosting instructions
First of all, I wanted to say how much I appreciate the Docker Compose file you've put together. 

I've been following the instructions at https://docs.graphql-hive.com/self-hosting/get-started and found that, once I give supertokens an API key longer than 20 characters (this also needs updating in your docs) and sign up/in, I'm always greeted with the following screen;

image

At the same time, a big red error is thrown in the server container. Here's the full log

```
[hive] is not enabled.
{"level":30,"time":1665948830035,"pid":1,"hostname":"67e225610f13","msg":"Service \"graphql-api\" is ready"}
{"level":30,"time":1665948830037,"pid":1,"hostname":"67e225610f13","msg":"Server listening at http://0.0.0.0:3001"}
{"level":20,"time":1665948830039,"pid":1,"hostname":"67e225610f13","msg":"Redis connection established"}
{"level":30,"time":1665948830040,"pid":1,"hostname":"67e225610f13","msg":"Redis connection ready"}
{"level":20,"time":1665948847104,"pid":1,"hostname":"67e225610f13","msg":"Running idempotent job (id=user:create:676c4f3b-00a5-4587-b754-e85355510037, traceId=baf81cc3d30a7) - (requestId=vJKdugLgQIqwQlRORDfgaA/0000000000)"}
{"level":20,"time":1665948847104,"pid":1,"hostname":"67e225610f13","msg":"Starting new job (id=user:create:676c4f3b-00a5-4587-b754-e85355510037, traceId=baf81cc3d30a7, attempt=1) - (requestId=vJKdugLgQIqwQlRORDfgaA/0000000000)"}
{"level":20,"time":1665948847109,"pid":1,"hostname":"67e225610f13","msg":"Fetching organizations - (requestId=vJKdugLgQIqwQlRORDfgaA/0000000001)"}
{"level":20,"time":1665948847110,"pid":1,"hostname":"67e225610f13","msg":"Running idempotent job (id=user:create:676c4f3b-00a5-4587-b754-e85355510037, traceId=9113001d92cc3) - (requestId=vJKdugLgQIqwQlRORDfgaA/0000000001)"}
{"level":20,"time":1665948847110,"pid":1,"hostname":"67e225610f13","msg":"Starting new job (id=user:create:676c4f3b-00a5-4587-b754-e85355510037, traceId=9113001d92cc3, attempt=1) - (requestId=vJKdugLgQIqwQlRORDfgaA/0000000001)"}
{"level":20,"time":1665948847110,"pid":1,"hostname":"67e225610f13","msg":"Job not found (id=user:create:676c4f3b-00a5-4587-b754-e85355510037, traceId=baf81cc3d30a7, attempt=1) - (requestId=vJKdugLgQIqwQlRORDfgaA/0000000000)"}
{"level":20,"time":1665948847110,"pid":1,"hostname":"67e225610f13","msg":"Trying to create a job (id=user:create:676c4f3b-00a5-4587-b754-e85355510037, traceId=baf81cc3d30a7, attempt=1) - (requestId=vJKdugLgQIqwQlRORDfgaA/0000000000)"}
{"level":20,"time":1665948847111,"pid":1,"hostname":"67e225610f13","msg":"Job not found (id=user:create:676c4f3b-00a5-4587-b754-e85355510037, traceId=9113001d92cc3, attempt=1) - (requestId=vJKdugLgQIqwQlRORDfgaA/0000000001)"}
{"level":20,"time":1665948847111,"pid":1,"hostname":"67e225610f13","msg":"Trying to create a job (id=user:create:676c4f3b-00a5-4587-b754-e85355510037, traceId=9113001d92cc3, attempt=1) - (requestId=vJKdugLgQIqwQlRORDfgaA/0000000001)"}
{"level":20,"time":1665948847123,"pid":1,"hostname":"67e225610f13","msg":"Job is pending (id=user:create:676c4f3b-00a5-4587-b754-e85355510037, traceId=9113001d92cc3) - (requestId=vJKdugLgQIqwQlRORDfgaA/0000000001)"}
{"level":20,"time":1665948847123,"pid":1,"hostname":"67e225610f13","msg":"Starting new job (id=user:create:676c4f3b-00a5-4587-b754-e85355510037, traceId=9113001d92cc3, attempt=2) - (requestId=vJKdugLgQIqwQlRORDfgaA/0000000001)"}
{"level":20,"time":1665948847124,"pid":1,"hostname":"67e225610f13","msg":"Job created (id=user:create:676c4f3b-00a5-4587-b754-e85355510037, traceId=baf81cc3d30a7, attempt=1) - (requestId=vJKdugLgQIqwQlRORDfgaA/0000000000)"}
{"level":20,"time":1665948847124,"pid":1,"hostname":"67e225610f13","msg":"Executing job (id=user:create:676c4f3b-00a5-4587-b754-e85355510037, traceId=baf81cc3d30a7, attempt=1) - (requestId=vJKdugLgQIqwQlRORDfgaA/0000000000)"}
{"level":20,"time":1665948847126,"pid":1,"hostname":"67e225610f13","msg":"Awaiting job (id=user:create:676c4f3b-00a5-4587-b754-e85355510037, traceId=9113001d92cc3, time=0) - (requestId=vJKdugLgQIqwQlRORDfgaA/0000000001)"}
{"context":{"package":"slonik","poolId":"Yt6aBGjkQMq4bQkC5tUtRA-0","logLevel":20,"stats":{"idleConnectionCount":0,"totalConnectionCount":1,"waitingRequestCount":0}},"message":"created a new client connection","sequence":"0","time":1665948847185,"version":"2.0.0"}
{"context":{"package":"slonik","poolId":"Yt6aBGjkQMq4bQkC5tUtRA-0","logLevel":20,"stats":{"idleConnectionCount":0,"totalConnectionCount":1,"waitingRequestCount":0}},"message":"client is checked out from the pool","sequence":"1","time":1665948847186,"version":"2.0.0"}
{"context":{"package":"slonik","poolId":"Yt6aBGjkQMq4bQkC5tUtRA-0","connectionId":"Yt6aBGjkQMq4bQkC5tUtRA-1","queryId":"Yt6aBGjkQMq4bQkC5tUtRA-2","logLevel":20,"sql":"\n        SELECT\n          *\n        FROM\n          public.\"users\"\n        WHERE\n          \"supertoken_user_id\" = $1\n        LIMIT 1\n      ","values":["676c4f3b-00a5-4587-b754-e85355510037"]},"message":"executing query","sequence":"2","time":1665948847187,"version":"2.0.0"}
{"context":{"package":"slonik","poolId":"Yt6aBGjkQMq4bQkC5tUtRA-0","connectionId":"Yt6aBGjkQMq4bQkC5tUtRA-1","queryId":"Yt6aBGjkQMq4bQkC5tUtRA-2","logLevel":20,"executionTime":"22ms","rowCount":null},"message":"query execution result","sequence":"3","time":1665948847211,"version":"2.0.0"}
{"level":30,"time":1665948847213,"pid":1,"hostname":"67e225610f13","msg":"Creating new user (input={\"superTokensUserId\":\"676c4f3b-00a5-4587-b754-e85355510037\",\"externalAuthUserId\":null,\"email\":\"milesbardon@gmail.com\"}) - (requestId=vJKdugLgQIqwQlRORDfgaA/0000000000)"}
{"context":{"package":"slonik","poolId":"Yt6aBGjkQMq4bQkC5tUtRA-0","logLevel":20,"stats":{"idleConnectionCount":0,"totalConnectionCount":1,"waitingRequestCount":0}},"message":"client is checked out from the pool","sequence":"4","time":1665948847215,"version":"2.0.0"}
{"context":{"package":"slonik","poolId":"Yt6aBGjkQMq4bQkC5tUtRA-0","connectionId":"Yt6aBGjkQMq4bQkC5tUtRA-3","queryId":"Yt6aBGjkQMq4bQkC5tUtRA-4","logLevel":20,"sql":"\n            INSERT INTO public.users\n              (\"email\", \"supertoken_user_id\", \"full_name\", \"display_name\", \"external_auth_user_id\")\n            VALUES\n              ($1, $2, $3, $4, $5)\n            RETURNING *\n          ","values":["milesbardon@gmail.com","676c4f3b-00a5-4587-b754-e85355510037","milesbardon","milesbardon",null]},"message":"executing query","sequence":"5","time":1665948847216,"version":"2.0.0"}
{"context":{"package":"slonik","poolId":"Yt6aBGjkQMq4bQkC5tUtRA-0","connectionId":"Yt6aBGjkQMq4bQkC5tUtRA-3","queryId":"Yt6aBGjkQMq4bQkC5tUtRA-4","logLevel":20,"executionTime":"18ms","rowCount":1},"message":"query execution result","sequence":"6","time":1665948847234,"version":"2.0.0"}
{"level":20,"time":1665948847235,"pid":1,"hostname":"67e225610f13","msg":"Marking job as completed (id=user:create:676c4f3b-00a5-4587-b754-e85355510037, traceId=baf81cc3d30a7, attempt=1) - (requestId=vJKdugLgQIqwQlRORDfgaA/0000000000)"}
{"level":20,"time":1665948847236,"pid":1,"hostname":"67e225610f13","msg":"Job completed (id=user:create:676c4f3b-00a5-4587-b754-e85355510037, traceId=baf81cc3d30a7) - (requestId=vJKdugLgQIqwQlRORDfgaA/0000000000)"}
Flag "startTransaction" is enabled but Sentry failed to find a transaction.
Try to create a transaction before GraphQL execution phase is started.
Flag "startTransaction" is enabled but Sentry failed to find a transaction.
Try to create a transaction before GraphQL execution phase is started.
❌   🧘 Yoga -   NotFoundError: Resource not found.
    at many (file:///usr/src/app/index.js:151706:15)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async createConnection (file:///usr/src/app/index.js:154218:18)
    at async Object.getOrganizations (file:///usr/src/app/index.js:204124:23)
    at async Promise.all (index 0)
    at async file:///usr/src/app/index.js:205917:17
    at async YogaServer.getResultForParams (file:///usr/src/app/index.js:208740:18)
    at async YogaServer.getResponse (file:///usr/src/app/index.js:208804:16)
    at async YogaServer.handle (file:///usr/src/app/index.js:208517:26)
    at async file:///usr/src/app/index.js:210566:24 {
  path: [ 'organizations' ],
  locations: [ { line: 2, column: 3 } ],
  extensions: [Object: null prototype] {}
} 
{"level":20,"time":1665948847629,"pid":1,"hostname":"67e225610f13","msg":"Resolving the job (id=user:create:676c4f3b-00a5-4587-b754-e85355510037, traceId=9113001d92cc3, attempt=2, status=COMPLETED) - (requestId=vJKdugLgQIqwQlRORDfgaA/0000000001)"}
{"context":{"package":"slonik","poolId":"Yt6aBGjkQMq4bQkC5tUtRA-0","logLevel":20,"stats":{"idleConnectionCount":0,"totalConnectionCount":1,"waitingRequestCount":0}},"message":"client is checked out from the pool","sequence":"7","time":1665948847630,"version":"2.0.0"}
{"context":{"package":"slonik","poolId":"Yt6aBGjkQMq4bQkC5tUtRA-0","connectionId":"Yt6aBGjkQMq4bQkC5tUtRA-5","queryId":"Yt6aBGjkQMq4bQkC5tUtRA-6","logLevel":20,"sql":"\n          SELECT o.*\n          FROM public.organizations as o\n          LEFT JOIN public.organization_member as om ON (om.organization_id = o.id)\n          WHERE om.user_id = $1\n          ORDER BY o.created_at DESC\n        ","values":["64980911-0d2d-40cc-9f0f-5b3327b52d4b"]},"message":"executing query","sequence":"8","time":1665948847632,"version":"2.0.0"}
{"context":{"package":"slonik","poolId":"Yt6aBGjkQMq4bQkC5tUtRA-0","connectionId":"Yt6aBGjkQMq4bQkC5tUtRA-5","queryId":"Yt6aBGjkQMq4bQkC5tUtRA-6","logLevel":20,"executionTime":"30ms","rowCount":null},"message":"query execution result","sequence":"9","time":1665948847662,"version":"2.0.0"}
{"context":{"package":"slonik","poolId":"Yt6aBGjkQMq4bQkC5tUtRA-0","connectionId":"Yt6aBGjkQMq4bQkC5tUtRA-5","logLevel":50,"queryId":"Yt6aBGjkQMq4bQkC5tUtRA-6"},"message":"NotFoundError","sequence":"10","time":1665948847662,"version":"2.0.0"}
{"context":{"package":"slonik","poolId":"Yt6aBGjkQMq4bQkC5tUtRA-0","logLevel":20,"stats":{"idleConnectionCount":0,"totalConnectionCount":0,"waitingRequestCount":0}},"message":"client connection is closed and removed from the client pool","sequence":"11","time":1665948847663,"version":"2.0.0"}
```

Would you be able to help me understand where I've gone wrong? The docker-compose config is exactly as specified in your documentation, without any changes (after running the `config` command to inject the env variables)

### PR DESCRIPTION
**Pull request by kamilkisiela**
Creates a special `ensureUser` trcp call that can be only accessed by `@hive/app`.
Removes the MessageBus, Redis and all that crap :)

The entire logic runs within a PG transaction and is triggered when a user sign in/up successfully. This way we do much less calls to DB.
Of course there could be a situation when something fails but I think it's less error prone than current implementation, at least less complex.

- [x] needs to be tested on staging

Closes #491

### CODE REVIEW

File: packages/services/api/package.json [hunk 1]
```diff
@@ -24,7 +24,6 @@
     "abort-controller": "3.0.0",
     "dataloader": "2.0.0",
     "date-fns": "2.25.0",
-    "emittery": "0.10.0",
```
Comment (n1ru4l) [inline packages/services/api/package.json:27]: 🥳

File: packages/web/app/src/config/backend-config.ts [hunk 2]
```diff
@@ -63,11 +67,87 @@ export const backendConfig = (): TypeInput => {
...
+            }
+
+            return newImplementation;
+          },
+          functions: originalImplementation => ({
+            ...originalImplementation,
+            /**
+             * These overrides are only relevant for the legacy Auth0 -> SuperTokens migration (period).
+             */
+            ...(env.auth.legacyAuth0 ? getAuth0Overrides(env.auth.legacyAuth0).functions(originalImplementation) : {}),
```
Comment (n1ru4l) [inline packages/web/app/src/config/backend-config.ts:148]: Can you please use this helper function for composing the overrides -

```ts
/**
 * Utility function for composing multiple (dynamic SuperTokens overrides).
 */
const composeSuperTokensOverrides = (overrides: Array<ThirdPartEmailPasswordTypeInput['override'] | null>) => ({
  apis: (
    originalImplementation: ReturnType<
      Exclude<Exclude<ThirdPartEmailPasswordTypeInput['override'], undefined>['apis'], undefined>
    >,
    builder: OverrideableBuilder<ThirdPartyEmailPasswordNode.APIInterface> | undefined
  ) => {
    let impl = originalImplementation;
    for (const override of overrides) {
      if (typeof override?.apis === 'function') {
        impl = override.apis(impl, builder);
      }
    }
    return impl;
  },
  functions: (
    originalImplementation: ReturnType<
      Exclude<Exclude<ThirdPartEmailPasswordTypeInput['override'], undefined>['functions'], undefined>
    >
  ) => {
    let impl = originalImplementation;
    for (const override of overrides) {
      if (typeof override?.functions === 'function') {
        impl = override.functions(impl);
      }
    }
    return impl;
  },
});
```

File: packages/services/server/.env.template [hunk 3]
```diff
@@ -33,6 +33,9 @@ ENCRYPTION_SECRET="97e4094d2463e71a981913cca4e56788"
 FEEDBACK_SLACK_TOKEN=""
 FEEDBACK_SLACK_CHANNEL="#hive"
 
+# Internal calls
+INTERNAL_ACCESS_SIGNATURE="123456"
```
Comment (n1ru4l) [inline packages/services/server/.env.template:37]: Nit: `AUTH_LEGACY_AUTH0_INTERNAL_API_KEY` could now also use this instead (just sayin)

**Review by n1ru4l — APPROVED** [review]
I approve this PR - pls fix the ~~tests~~ build and consider my feedback :)

File: packages/services/server/.env.template (hunk 3 above)
Comment (kamilkisiela) [reply packages/services/server/.env.template:37]: `INTERNAL_ACCESS_SIGNATURE` is something I'm going to remove, since the API calls are server-side.

**Pull request merged**